### PR TITLE
feature: system

### PR DESCRIPTION
**Pull request by alimoncul**

### CODE REVIEW

File: device/system/system.ios.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  }
+  get biometricsAvailable() {
+    const context = new __SF_LAContext();
+    return context.canEvaluatePolicy();
+  }
+  vibrate() {
+    __SF_UIDevice.vibrate();
+  }
+  validateBiometric(params) {
+    const context = new __SF_LAContext();
```
Comment (cenkce) [inline device/system/system.ios.ts:84]: params tipini verelim

File: device/system/system.ios.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    context.evaluatePolicy(params.message, params.onSuccess, params.onError);
+  }
+  validateFingerPrint(params) {
```
Comment (cenkce) [inline device/system/system.ios.ts:87]: params tipini verelim

File: device/system/system.ios.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    this.validateBiometric(params);
+  }
+  isApplicationInstalled(packageName) {
+    const url = __SF_NSURL.URLWithString(packageName);
```
Comment (cenkce) [inline device/system/system.ios.ts:91]: params tipini verelim

File: device/system/system.android.ts [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    biometricPrompt.setSubTitle(subTitle);
+    biometricPrompt.setCancelButtonText(cancelButtonText);
+    biometricPrompt.setConfirmationRequired(confirmationRequired);
+
+    biometricPrompt.authenticate();
+  }
+  get biometricType() {
+    return SFBiometricPrompt.getBiometricType(AndroidConfig.activity);
+  }
+  validateFingerPrint(params) {
```
Comment (cenkce) [inline device/system/system.android.ts:154]: params tipini verelim

File: device/system/system.android.ts [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    if (AndroidConfig.sdkVersion >= AndroidConfig.SDK.SDK_MARSHMALLOW) {
+      const fingerprintManager = AndroidConfig.getSystemService(FINGERPRINT_SERVICE, FINGERPRINT_MANAGER);
+      return fingerprintManager && fingerprintManager.isHardwareDetected() && fingerprintManager.hasEnrolledFingerprints();
+    }
+    return false;
+  }
+  get biometricsAvailable() {
+    return SFBiometricPrompt.getBiometricType(AndroidConfig.activity) === this.BiometryType.BIOMETRICS;
+  }
+  validateBiometric(params) {
```
Comment (cenkce) [inline device/system/system.android.ts:126]: params tipini verelim

**Review by cenkce — CHANGES_REQUESTED** [review]

**Comment by alimoncul** [comment]
@cenkce updated

**Pull request merged**